File: ECnotes/Sem1/Labs/LivingExamples/VB/Ch16ProductList/App_Code/ProductDB.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports System.Data
Imports System.Data.SqlClient

<DataObject(True)> _
Public Class ProductDB

    <DataObjectMethod(DataObjectMethodType.Select)> _
    Public Shared Function GetAllCategories() As IEnumerable
        Dim sel As String = "SELECT CategoryID, LongName " _
            & "FROM Categories ORDER BY LongName"
        Dim cmd As SqlCommand = _
            New SqlCommand(sel, New SqlConnection(GetConnectionString))
        cmd.Connection.Open()
        Dim dr As SqlDataReader = cmd.ExecuteReader(CommandBehavior.CloseConnection)
        Return dr
    End Function

    <DataObjectMethod(DataObjectMethodType.Select)> _
    Public Shared Function GetProductsByCategory(ByVal CategoryID As String) _
            As IEnumerable
        Dim sel As String = "SELECT ProductID, Name, " _
            & "UnitPrice, OnHand " _
            & "FROM Products " _
            & "WHERE CategoryID = @CategoryID " _
            & "ORDER BY ProductID"
        Dim cmd As SqlCommand = _
            New SqlCommand(sel, New SqlConnection(GetConnectionString))
        cmd.Parameters.AddWithValue("CategoryID", CategoryID)
        cmd.Connection.Open()
        Dim dr As SqlDataReader = cmd.ExecuteReader(CommandBehavior.CloseConnection)
        Return dr
    End Function

    Private Shared Function GetConnectionString() As String
        Return ConfigurationManager.ConnectionStrings _
            ("HalloweenConnection").ConnectionString
    End Function

End Class

File: ECnotes/Sem1/Labs/LivingExamples/CS/Ch16ProductList/App_Code/ProductDB.cs
using System;
using System.Collections;
using System.Data;
using System.Data.SqlClient;
using System.Configuration;
using System.ComponentModel;

[DataObject(true)]
public static class ProductDB
{
    [DataObjectMethod(DataObjectMethodType.Select)]
    public static IEnumerable GetAllCategories()
    {
        string sel = "SELECT CategoryID, LongName "
                   + "FROM Categories ORDER BY LongName";

        SqlCommand cmd = new SqlCommand(sel, new SqlConnection(GetConnectionString()));

        cmd.Connection.Open();

        SqlDataReader dr = cmd.ExecuteReader(CommandBehavior.CloseConnection);

        return dr;
    }

    [DataObjectMethod(DataObjectMethodType.Select)]
    public static IEnumerable GetProductsByCategory(string CategoryID)
    {
        string sel = "SELECT ProductID, Name, "
                   + "       UnitPrice, OnHand "
                   + "FROM Products "
                   + "WHERE CategoryID = @CategoryID "
                   + "ORDER BY ProductID";

        SqlCommand cmd = new SqlCommand(sel, new SqlConnection(GetConnectionString()));

        cmd.Parameters.AddWithValue("CategoryID", CategoryID);

        cmd.Connection.Open();

        SqlDataReader dr = cmd.ExecuteReader(CommandBehavior.CloseConnection);

        return dr;
    }

    private static string GetConnectionString()
    {
        return ConfigurationManager.ConnectionStrings["HalloweenConnectionString"].ConnectionString;
    }
}
